### PR TITLE
Jetpack connect: add a feature flag to allow disabling site questions

### PR DESCRIPTION
**Pull request by simison**
#### Changes proposed in this Pull Request

* Put site questions in Jetpack connect flow behind a feature flag on all environments; the flow should stay visible everywhere after this change.
* ~Disable related E2E tests (convo p1562574334001800-slack-e2e-testing-discuss)~

Next up we'd flip these feature flags off to disable the steps in all environments. (PR: https://github.com/Automattic/wp-calypso/pull/34620)

Context paObgF-k1-p2

Jetpack connect steps

#### Testing instructions
- Create a new Jetpack site and connect it to .com using localhost.
- Can still see extra questions?

### CODE REVIEW

**Comment by Stojdza** [comment]
@simison You can merge the latest changes from master. It should affect jetpack tests and a8c/valid-pr should pass. More info: p4TIVU-9g0-p2

File: config/development.json [hunk 1]
```diff
@@ -78,6 +78,7 @@
 		"jetpack/connect/remote-install": true,
 		"jetpack/connect/mobile-app-flow": true,
 		"jetpack/connect/woocommerce": true,
+		"jetpack/connect/site-questions": true,
```
Comment (simison) [inline config/development.json]: We'll need to add this flag to all other env configs as well to ensure that the steps stay visible everywhere when this PR is merged.

**Comment by simison** [comment]
@jeffgolenski and I agreed that we can merge this, but with flags __enabled__, so that onboarding design sprint can happen first. During the design sprint, they could then turn the flag off to test the flow without these steps.

After the spring, we can hide these steps by switching the feature flag off.

**Comment by bsessions85** [comment]
@simison e2e test changes look good, but do we need to remove those changes until we turn the feature flag off?

**Comment by simison** [comment]
@bsessions85 you're right. I've enabled tests again in https://github.com/Automattic/wp-calypso/pull/34506/commits/eb85423483e9da71e9fdc9c6efeaab1ffd78e87d

**Comment by simison** [comment]
Follow up PR after this PR is merged: https://github.com/Automattic/wp-calypso/pull/34620 (to actually disable the flow with these feature flags)

**Review by sirreal — APPROVED** [review]
Tested, compared with production I see the exact same flow on calypso.live 👍

**Pull request merged**